### PR TITLE
Drop PHF from TPM dependency

### PR DESCRIPTION
**Pull request by liuw**
See https://github.com/cloud-hypervisor/cloud-hypervisor/issues/4873

This drops the number of dependencies from 129 to 120.

No change in build time.

### CODE REVIEW

**Review by rbradford — APPROVED** [review]
Going further can't you replace the strings with an enum as they're not part of the protocol anyway?

**Review by sboeuf — APPROVED** [review]

**Comment by praveen-pk** [comment]
Thanks for driving this @liuw. I started working on replacing phf with bitflags. This implementation looks much simpler.

**Review by praveen-pk — APPROVED** [review]

**Comment by liuw** [comment]
Don't merge this yet. If I find some time today I may even write a version that removes all the strings.

**Review by rbradford — APPROVED** [review]

File: devices/src/tpm.rs [hunk 1]
```diff
@@ -28,57 +27,132 @@ pub enum Error {
...
+#[allow(dead_code)]
+enum LocStateFields {
+    TpmEstablished,
+    LocAssigned,
+    ActiveLocality,
+    Reserved,
+    TpmRegValidSts,
+}
+
+enum LocStsFields {
```
Comment (praveen-pk) [inline devices/src/tpm.rs:39]: don't you need the `dead_code` lint for rest of the enums here?

Comment (liuw) [reply devices/src/tpm.rs:39]: Yes

**Comment by liuw** [comment]
The failure is not related to my change. Not sure why Clippy is triggered like that...

**Comment by rbradford** [comment]
@liuw I expect the toolchain just got updated with beta updated.

**Pull request merged**